cmd/data-crawler.go

```diff
@@ -738,8 +745,28 @@ func (i *crawlItem) applyActions(ctx context.Context, o ObjectLayer, meta action
 			}
 		}
 		opts.VersionID = obj.VersionID
+		if obj.Transitioned {
```

Comment:
I think this code could be moved to little later, 
```
if obj.Transitioned {
   deleteTransitionedObject()
} else {
   o.DeleteObject()
}
```

Reply:
done